src/analyses/goto_check.cpp

```diff
@@ -900,23 +902,56 @@ void goto_checkt::pointer_overflow_check(
   if(!enable_pointer_overflow_check)
     return;
 
-  if(expr.id()==ID_plus ||
-     expr.id()==ID_minus)
+  if(expr.id() != ID_plus && expr.id() != ID_minus)
+    return;
+
+  if(expr.operands().size() > 2)
   {
-    if(expr.operands().size()==2)
+    // The overflow checks are binary!
+    // We break these up.
+
+    for(std::size_t i = 1; i < expr.operands().size(); i++)
     {
-      exprt overflow("overflow-"+expr.id_string(), bool_typet());
-      overflow.operands()=expr.operands();
+      exprt tmp;
 
-      add_guarded_claim(
-        not_exprt(overflow),
-        "pointer arithmetic overflow on "+expr.id_string(),
-        "overflow",
-        expr.find_source_location(),
-        expr,
-        guard);
+      if(i == 1)
+        tmp = expr.op0();
+      else
+      {
+        tmp = expr;
+        tmp.operands().resize(i);
+      }
+
+      exprt tmp2 = expr;
+      tmp2.operands().resize(2);
+      tmp2.op0() = tmp;
+      tmp2.op1() = expr.operands()[i];
+
+      pointer_overflow_check(tmp2, guard);
     }
   }
+  else
+  {
+    DATA_INVARIANT(
+      expr.operands().size() == 2,
+      "overflow-" + expr.id_string() + " takes 2 operands");
+
+    exprt overflow("overflow-" + expr.id_string(), bool_typet());
+    overflow.operands() = expr.operands();
+
+    // this is overflow over integers
```

Comment:
How about `// check for address space overflow`, since I guess the point is we're checking whether we wrap the VAS?